### DISCUSSION TITLE
Script Was Working Thursday 7-29-21 But Wasn't Working Yesterday 8-3-21 Or Today 8-4-21

### DISCUSSION THREAD

**Discussion by bffive** [grahampugh/erase-install#115] [Q&A] [answered]
Pushing this from JAMF using the following after parameters on each line. 
--force-curl
--erase
--os=11

Also tried --os=11.5
and yesterday --os=10.15 (which is what had worked last thursday.)

Below is the error output

<html>
<body>

Attempt 1 Today at 11:53 AM
--
Executing Policy Install macOS 11 Big Sur (ERASES EVERYTHING!)
Running script Erase-Install.sh...
Script exit code: 1
Script result:    [erase-install] v0.20.1 script execution started: Wed Aug  4 11:46:49 EDT 2021    [erase-install] Caffeinating this script (pid=1084) 45000000    [free_space_check] OK - 580560660 KB free disk space detected    [erase-install] Looking for existing installer    [find_existing_installer] No valid installer found.    [erase-install] Starting download process    [erase-install] Opening jamfHelper download message (language=en)    [get_installinstallmacos] Downloading installinstallmacos.py...    [get_installinstallmacos] installinstallmacos.py is in /Library/Management/erase-install    [run_installinstallmacos] Checking that selected OS 11 is available     [run_installinstallmacos] This command is now being run:     installinstallmacos.py --workdir=/Library/Management/erase-install --ignore-cache --raw --os=11 --validate  Could not replicate https://swdist.apple.com/content/downloads/20/55/001-51042-A_2EJTJOSUC2/rsvf13iphg5lvcqcysqcarv8cvddq8igek/001-51042.English.dist: Command '[u'/usr/bin/curl', u'-sfL', u'--create-dirs', u'-o', u'/Library/Management/erase-install/content/downloads/20/55/001-51042-A_2EJTJOSUC2/rsvf13iphg5lvcqcysqcarv8cvddq8igek/001-51042.English.dist', u'-w', u'%{http_code}', u'--compressed', 'https://swdist.apple.com/content/downloads/20/55/001-51042-A_2EJTJOSUC2/rsvf13iphg5lvcqcysqcarv8cvddq8igek/001-51042.English.dist']' returned non-zero exit status 7 Could not replicate https://swdist.apple.com/content/downloads/41/23/061-26578-A_FF0C6M87LS/ylvl0phx6uy3i444qt9p57n8wz0xnylmd9/061-26578.English.dist: Command '[u'/usr/bin/curl', u'-sfL', u'--create-dirs', u'-o', u'/Library/Management/erase-install/content/downloads/41/23/061-26578-A_FF0C6M87LS/ylvl0phx6uy3i444qt9p57n8wz0xnylmd9/061-26578.English.dist', u'-w', u'%{http_code}', u'--compressed', 'https://swdist.apple.com/content/downloads/41/23/061-26578-A_FF0C6M87LS/ylvl0phx6uy3i444qt9p57n8wz0xnylmd9/061-26578.English.dist']' returned non-zero exit status 7 Could not replicate https://swdist.apple.com/content/downloads/23/45/001-36735-A_IRCLJ5YYKF/z8vka6xfykydycui5b7atpc52btxmbffdn/001-36735.English.dist: Command '[u'/usr/bin/curl', u'-sfL', u'--create-dirs', u'-o', u'/Library/Management/erase-install/content/downloads/23/45/001-36735-A_IRCLJ5YYKF/z8vka6xfykydycui5b7atpc52btxmbffdn/001-36735.English.dist', u'-w', u'%{http_code}', u'--compressed', 'https://swdist.apple.com/content/downloads/23/45/001-36735-A_IRCLJ5YYKF/z8vka6xfykydycui5b7atpc52btxmbffdn/001-36735.English.dist']' returned non-zero exit status 7 Could not replicate https://swdist.apple.com/content/downloads/08/44/041-90855-A_452WLVQACP/mzfvoi59bljh3w4w4lvwenfk0r1a2f7212/041-90855.English.dist: Command '[u'/usr/bin/curl', u'-sfL', u'--create-dirs', u'-o', u'/Library/Management/erase-install/content/downloads/08/44/041-90855-A_452WLVQACP/mzfvoi59bljh3w4w4lvwenfk0r1a2f7212/041-90855.English.dist', u'-w', u'%{http_code}', u'--compressed', 'https://swdist.apple.com/content/downloads/08/44/041-90855-A_452WLVQACP/mzfvoi59bljh3w4w4lvwenfk0r1a2f7212/041-90855.English.dist']' returned non-zero exit status 7 Could not replicate https://swdist.apple.com/content/downloads/10/06/001-36801-A_9MBEQ97Z9M/672a1q4u8978ize3s15ygivp2dqwmk10qb/001-36801.English.dist: Command '[u'/usr/bin/curl', u'-sfL', u'--create-dirs', u'-o', u'/Library/Management/erase-install/content/downloads/10/06/001-36801-A_9MBEQ97Z9M/672a1q4u8978ize3s15ygivp2dqwmk10qb/001-36801.English.dist', u'-w', u'%{http_code}', u'--compressed', 'https://swdist.apple.com/content/downloads/10/06/001-36801-A_9MBEQ97Z9M/672a1q4u8978ize3s15ygivp2dqwmk10qb/001-36801.English.dist']' returned non-zero exit status 7  installinstallmacos.py - get macOS installers from the Apple software catalog  This Mac: Model Identifier : iMac18,1  Board ID         : Mac-4B682C642B45593E OS Version       : 10.14.6 Build ID         : 18G95   #  ProductID       Version    Build    Post Date   Title                            1  071-71342       11.5       20G71    2021-07-21  macOS Big Sur                    2  001-68446       10.15.7    19H15    2020-11-11  macOS Catalina                   3  001-04366       10.15.4    19E2269  2020-05-04  macOS Catalina                  Traceback (most recent call last):   File "/Library/Management/erase-install/installinstallmacos.py", line 1323, in <module>     main()   File "/Library/Management/erase-install/installinstallmacos.py", line 947, in main     if board_id and product_info[product_id]["BoardIDs"]: KeyError: u'BoardIDs'    [run_installinstallmacos] Error obtaining valid installer. Cannot continue.    [erase-install] 'jamfHelper' ended
Error running script: return code was 1.

</body>
</html>

**Comment by grahampugh** [comment, marked as answer]
That's just network issues.

**Reply by bffive** [reply]
Oh wow, so it's on my end?

**Reply by grahampugh** [reply]
Yep. You can see the curl timeouts in the output.

**Reply by bffive** [reply]
Thanks so much. I'll try to troubleshoot as to why since it probably means there's a much bigger problem going on!

**Reply by bffive** [reply]
I was able to hook a device off the network, and sure enough, worked right away! Thanks again for the help!

**Comment by grahampugh** [comment]
By the way, --os=11.5 won't work as that is a minor version. Use --os=11 or --version=11.5

**Reply by bffive** [reply]
Oh sweet, thanks so much for the clarification. I'm new to your script, and well scripts in general and I'm very slow on the uptake. (Bad combo) If I use --os=11, does it automatically get the latest verion, eg 11.5 as of today? Or would it install the first version of OS 11 that was released?

**Reply by grahampugh** [reply]
It'll release the latest available version of macOS 11.

**Reply by bffive** [reply]
That's amazing. You're the best, thanks!
